src/platforms/platform.rs

```
@@ -92,35 +78,56 @@ pub trait OverlayHandler {
     fn overlay_fs_path(&self) -> Result<&Path, FpgadError>;
 }
 
-fn match_platform_string(platform_string: &str) -> Result<PlatformType, FpgadError> {
-    for (platform, substrs) in PLATFORM_SUBSTRINGS {
-        let platform_subs: Vec<&str> = platform_string.split(',').collect();
+pub trait Platform {
+    /// creates and inits an Fpga if not present otherwise gets the instance
+    fn fpga(&self, device_handle: &str) -> Result<&dyn Fpga, FpgadError>;
+    /// creates and inits an OverlayHandler if not present otherwise gets the instance
+    fn overlay_handler(&self, overlay_handle: &str) -> Result<&(dyn OverlayHandler), FpgadError>;
+}
+
+fn match_platform_string(platform_string: &str) -> Result<Box<dyn Platform>, FpgadError> {
+    let registry = PLATFORM_REGISTRY
+        .get()
+        .expect("PLATFORM_REGISTRY is not initialized")
+        .lock()
+        .expect("Failed to lock PLATFORM_REGISTRY");
```

Can we propagate the error back to the caller here?

---

```suggestion
    let registry = PLATFORM_REGISTRY.get()?.lock()?;
```